tests/base/test_components.py

```diff
@@ -179,6 +180,140 @@ def test_init_ComponentError_given_duplicated_component_name(self):
         with pytest.raises(ComponentError):
             Component("Sector 1", parent=parent)
 
+    def test_copy_does_fully_copy(self):
+        parent = Component("Parent")
+        child1 = Component("Child1", parent=parent)
+        child2 = PhysicalComponent(
+            "Child2", parent=parent, shape="A shape", material="A material"
+        )
+        Component("GrandchildAA", parent=child1)
+        Component("GrandchildAB", parent=child1)
+        PhysicalComponent(
+            "GrandchildBA", parent=child2, shape="B shape", material="B material"
+        )
+        Component("GrandchildBB", parent=child2)
+
+        parent_copy = parent.copy()
+
+        def get_comp_copy_and_compare(comp: Component):
+            cpy = parent_copy.get_component(comp.name)
+
+            # test instance
+            assert cpy is not None
+            assert cpy is not comp
+
+            # test parent
+            if cpy.parent:
+                # assert cpy.parent is not comp.parent
+                assert cpy.parent.name == comp.parent.name
+
+            # test children
+            if cpy.children:
+                comp_children_names = [c.name for c in comp.children]
+                for c in cpy.children:
+                    assert c.name in comp_children_names
+
+            # test properties
+            if type(comp) is PhysicalComponent:
+                assert type(cpy) is PhysicalComponent
```

Comment:
In this case, I would argue that because I am specifically testing if the type is `PhysicalComponent` it's better to use the `type(x) is` as I'm not interested in inheritance.

Your change is of course correct because nothing sub-classes `PhysicalComponent` but if something were to, the succeeding checks wouldn't really make sense (although would probably still work).

I'm happy to make this change if it conforms more to the general style though.